File: 2015/day2/part2.swift
import Foundation

func part2(_ input: String) -> Int {
	let inputSplit = input.components(separatedBy: "\n")
	let result = inputSplit.map { (line: String) -> Int in
		let dimensions = line.components(separatedBy: "x").map { Int(String($0)) ?? 0 }.sorted()
		// perimeter of smallest side (2 * w + 2 * h) + volume (l * w * h)
		return 2 * dimensions[0] + 2 * dimensions[1] +
			dimensions[0] * dimensions[1] * dimensions[2]
	}.reduce(0, +)
}

File: 2015/day2/part2.py
def part2(dimensions):
    ret = 0
    for line in dimensions.split('\n'):
        line_sorted = sorted(list(map(int, line.split('x'))))
        ret += line_sorted[0] * 2 + line_sorted[1] * 2 + \
            line_sorted[0] * line_sorted[1] * line_sorted[2]
    return ret
